.github/workflows/consumer_contract_tests.yml

```
@@ -0,0 +1,145 @@
+            GITHUB_SHA=${{ github.event.pull_request.head.sha }}
+          elif [[ "$GITHUB_EVENT_NAME" == "merge_group" ]]; then
+            GITHUB_REF=refs/heads/${{ github.head_ref }}
+          else
+            echo "Failed to extract branch information"
+            exit 1
+          fi
+          echo "repo-branch=${GITHUB_REF/refs\/heads\//""}" >> $GITHUB_OUTPUT
+          echo "repo-version=${GITHUB_SHA}" >> $GITHUB_OUTPUT
+          echo "fork=${FORK}" >> $GITHUB_OUTPUT
+
+      - name: Is PR triggered by forked repo?
+        if: ${{ steps.extract-branch.outputs.fork == 'true' }}
+        run: |
+          echo "PR was triggered by forked repo"
+
+      - name: Echo repo and branch information
+        run: |
+          echo "repo-owner=${{ github.repository_owner }}"
+          echo "repo-name=${{ github.event.repository.name }}"
+          echo "repo-branch=${{ steps.extract-branch.outputs.repo-branch }}"
+          echo "repo-version=${{ steps.extract-branch.outputs.repo-version }}"
+
+  cromwell-consumer-contract-tests:
+    runs-on: ubuntu-latest
+    needs: [init-github-context]
+    outputs:
+      pact-b64: ${{ steps.encode-pact.outputs.pact-b64 }}
+
+    steps:
+      - uses: actions/checkout@v3
+      - name: Run consumer tests
+        run: |
+          docker run --rm -v $PWD:/working \
+                          -v jar-cache:/root/.ivy \
+                          -v jar-cache:/root/.ivy2 \
+                          -w /working \
+                          sbtscala/scala-sbt:openjdk-17.0.2_1.7.2_2.13.10 \
+                          sbt "project pact4s" clean test
+
+      - name: Output consumer contract as non-breaking base64 string
+        id: encode-pact
+        run: |
+          cd pact4s
+          NON_BREAKING_B64=$(cat target/pacts/cromwell-consumer-sam-provider.json | base64 -w 0)
+          echo "pact-b64=${NON_BREAKING_B64}" >> $GITHUB_OUTPUT
+
+  # Prevent untrusted sources from using PRs to publish contracts
+  # since access to secrets is not allowed.
+  publish-contracts:
+    runs-on: ubuntu-latest
+    if: ${{ needs.init-github-context.outputs.fork == 'false' }}
+    needs: [init-github-context, cromwell-consumer-contract-tests]
+    steps:
+      - name: Dispatch to terra-github-workflows
+        uses: broadinstitute/workflow-dispatch@v3
+        with:
+          workflow: .github/workflows/publish-contracts.yaml
+          repo: broadinstitute/terra-github-workflows
+          ref: refs/heads/main
+          token: ${{ secrets.BROADBOT_TOKEN }} # github token for access to kick off a job in the private repo
+          inputs: '{ "pact-b64": "${{ needs.cromwell-consumer-contract-tests.outputs.pact-b64 }}", "repo-owner": "${{ github.repository_owner }}", "repo-name": "${{ github.event.repository.name }}", "repo-branch": "${{ needs.init-github-context.outputs.repo-branch }}" }'
+
+  can-i-deploy:
+    runs-on: ubuntu-latest
+    if: ${{ needs.init-github-context.outputs.fork == 'false' }}
```

You can disable `can-i-deploy` and publish the `cromwell` contract to pact broker if you like. Later, when your `drshub` provider test in this [PR](https://github.com/DataBiosphere/terra-drs-hub/pull/60) is done you can add a webhook (essentially just like [https://github.com/broadinstitute/sam/blob/develop/.github/workflows/verify_consumer_pacts.yml](https://github.com/broadinstitute/sam/blob/develop/.github/workflows/verify_consumer_pacts.yml) and you can reenable `can-i-deploy` again. This is just a suggestion.

```suggestion
    if: ${{ needs.init-github-context.outputs.fork == 'false' && false  }}
```

---

That sounds good thanks